app/src/main/java/com/example/whatshouldiread/ui/profile/ProfileViewModel.kt

```diff
@@ -10,4 +10,31 @@ class ProfileViewModel : ViewModel() {
         value = "This is notifications Fragment"
     }
     val text: LiveData<String> = _text
+}
+
+
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    ...
+    handleIntent(intent)
+}
+
+override fun onNewIntent(intent: Intent) {
+    super.onNewIntent(intent)
+    handleIntent(intent)
+}
+
+private fun handleIntent(intent: Intent) {
+    val appLinkAction = intent.action
+    val appLinkData: Uri? = intent.data
+    if (Intent.ACTION_VIEW == appLinkAction) {
+        appLinkData?.lastPathSegment?.also { recipeId ->
+            Uri.parse("content://com.recipe_app/recipe/")
+                .buildUpon()
+                .appendPath(recipeId)
+                .build().also { appData ->
+                    showRecipe(appData)
+                }
+        }
+    }
 }
```

Comment:
Code Review 2

Analysis of the program:
• Was the program available in UC Github on time? Yes
• Is the program documented/commented well enough for you to understand? Yes
• Does the program compile? Yes
• Rationale behind your changes? Functionality.

Changes I made:
• Added an linkage functionality to ProfileViewModel.kt. There was no way to link apps prior to this addition. Great way to add on to any ideas to handle incoming links.

Things I learned:
•Click Select Activity from the App Links Assistant.
 Select an activity from the list and click Insert Code.
 The App Links Assistant adds code to your activity's Java file, similar to the following ( Please note: Currently the App Links Assistant does not support Kotlin so you will need to add this code manually):